### ISSUE TITLE
Stale oc_properties values in the wild

### ISSUE THREAD

**Issue by tomneedham** [owncloud/core#34734]
ownCloud 10.x - Server

Sometimes apps / plugins etc add properties into `oc_properties` - which relate to a `fileid` on the `oc_filecache` table. There is no foreign key with cascading delete - so deleted in the filecache can cause stale entries in the `oc_properties` table. Individual plugins are then resposible for implementing a `OCP\Files\NotFoundExcepetion` check which doesn't always happen / insn't in place yet.

You can spot stale entries with thie query:

`SELECT * FROM oc_properties p WHERE NOT EXISTS (SELECT null FROM oc_filecache f WHERE f.fileid = p.fileid)`

Maybe we should run this on a few instnaces to see what `propertyname` values we have? 

I've seen at least `upload-time` which was left of over `holding_period`. And `{http://owncloud.org/ns}role ` with a null `fileid` ?!

We could cleanup this in a maintenance step, or get individual use cases to cleanup their entries in repair steps?

**Comment by tomneedham** [comment]
Related to https://github.com/owncloud/core/issues/13143

**Comment by PVince81** [comment]
isn't this solved with https://github.com/owncloud/core/issues/33413 that was released with 10.1.0 ?

**Comment by PVince81** [comment]
foreign keys would be nice though... we started adding some for the oc_persistent_locks properties

**Comment by tomneedham** [comment]
> isn't this solved with #33413 that was released with 10.1.0 ?

Yes - but existing stale values. Maintennace step to cleanup would be nice. 

Where does `{http://owncloud.org/ns}role` come from, why does it have a NULL fileid, and present multiple times?

**Comment by DeepDiver1975** [comment]
as designed - bad designed - but as designed

solution: build a occ command for clean up

**Comment by DeepDiver1975** [comment]
🤖 Automated issue triage by Claude Code review agent.

This issue was opened before 2020 and has had no activity since then, so we are closing it as outdated. If this is still relevant on a current ownCloud release, please **reopen** (or open a fresh issue) with up-to-date details — thank you!
